web/api/servers.go

```diff
@@ -721,6 +722,50 @@ func removeServerUser(c *gin.Context) {
 	c.Status(http.StatusNoContent)
 }
 
+// @Summary Rename server
+// @Description Renames a server
+// @Accept json
+// @Produce json
+// @Success 200
+// @Failure 400 {object} response.Error
+// @Failure 403 {object} response.Error
+// @Failure 404 {object} response.Error
+// @Failure 500 {object} response.Error
+// @Param id path string false "Server ID"
+// @Param name path string false "Server Name"
+// @Router /api/servers/{id}/name [post]
+func renameServer(c *gin.Context) {
+	var err error
+	db := middleware.GetDatabase(c)
+	ss := &services.Server{DB: db}
+	serverId := c.Param("serverId")
+	if serverId == "" {
+		c.AbortWithStatus(http.StatusNotFound)
+		return
+	}
+	name := c.Param("name")
+	if name == "" {
+		c.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	server, err := ss.Get(serverId)
```

Comment:
Can get the server from the context, saves on a db call, but isn't too bad
https://github.com/PufferPanel/PufferPanel/blob/1beffb5b8a6131a5e69679d433d7298839047ea0/web/api/servers.go#L392-L403